File: src/main/java/com/aliiensmp/aliienResize/Config/Records/CachedActionItem.java
package com.aliiensmp.aliienResize.Config.Records;

import com.aliiensmp.aliienResize.Menus.MenuAction;
import org.bukkit.inventory.ItemStack;

public record CachedActionItem(
        int slot,
        MenuAction action,
        int targetPage,
        ItemStack item
) {}

File: src/main/kotlin/com/aliiensmp/aliienResize/config/data/CachedActionItem.kt
package com.aliiensmp.aliienResize.config.data

import com.aliiensmp.aliienResize.menus.MenuAction
import org.bukkit.inventory.ItemStack

data class CachedActionItem(
    val slot: Int,
    val action: MenuAction,
    val targetPage: Int,
    val item: ItemStack
)
